### PR TITLE
Add Vial support for idobao/montex/v2

### PR DESCRIPTION
**Pull request by OTLabs**

### CODE REVIEW

File: keyboards/idobao/montex/v2/keymaps/vial/vial.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    {"name": "", "title": "{undefined}", "shortName": ""},
+    {"name": "", "title": "{undefined}", "shortName": ""},
+    {"name": "", "title": "{undefined}", "shortName": ""},
+    {
+      "name": "Toggle NumLk LED",
+      "title": "Toggle NumLock LED Feature",
+      "shortName": "NumLk LED"
+    },
+    {"name": "Version", "title": "Type Firmware Version", "shortName": "Ver?"}
+  ],
```
Comment (lesshonor) [inline keyboards/idobao/montex/v2/keymaps/vial/vial.json]:
```suggestion
```
The custom keycodes aren't programmed at all and the lighting section is VIA cut-paste. If you don't want to just get rid of this, [reading the documentation](https://get.vial.today/docs/lighting.html#rgb-matrix--vialrgb) can help you fix it.

**Review by lesshonor — CHANGES_REQUESTED** [review]
[This doesn't build](https://github.com/lesshonor/vial-qmk/actions/runs/6223944703/job/16891083724#step:4:18).

File: keyboards/idobao/montex/v2/keymaps/vial/vial.json (hunk 1 above)
Comment (OTLabs) [reply keyboards/idobao/montex/v2/keymaps/vial/vial.json]: Sure, let me check, thank you

**Comment by OTLabs** [comment]
> [This doesn't build](https://github.com/lesshonor/vial-qmk/actions/runs/6223944703/job/16891083724#step:4:18).

Ouch. Right now local build system reports 4 free bytes:
```* The firmware size is approaching the maximum - 28668/28672 (99%, 4 bytes free)```

What should I change to get same result as yours? The difference is quite notable. I see how to save 100 bytes more, but 202 would be complicated.

**Comment by lesshonor** [comment]
> What should I change to get same result as yours?

I'm guessing you aren't using the same version of `avr-gcc` (8.x).

> I see how to save 100 bytes more, but 202 would be complicated.

- https://get.vial.today/docs/firmware-size.html
- https://docs.qmk.fm/#/squeezing_avr

**Comment by OTLabs** [comment]
> I'm guessing you aren't using the same version of `avr-gcc` (8.x).

Looks like I use `avr-gcc` 8.5.0:
```
[OTLabs@DESKTOP ~]$ qmk doctor                                    
Ψ QMK Doctor is checking your environment.                              
Ψ CLI version: 1.1.1                                                    
Ψ QMK home: C:/Users/OTLabs/qmk_firmware                                  
Ψ Detected Windows 10 (10.0.19045).                                     
Ψ QMK MSYS version: 1.7.2                                               
Ψ Git branch: master                                                    
Ψ Repo version: 0.22.2                                                  
Ψ - Latest master: 2023-09-09 12:15:57 -0700 (9a29ddf6af) -- dnworks FRLTKL Community Layout support (#21922)                                   
Ψ - Latest upstream/master: 2023-09-18 19:55:51 +0100 (16f2a3e2f8) -- Add missing gpio include to ps2 vendor (#22063)                           
Ψ - Latest upstream/develop: None                                       
Ψ - Common ancestor with upstream/master: 2023-09-09 12:15:57 -0700 (9a29ddf6af) -- dnworks FRLTKL Community Layout support (#21922)
Ψ - Common ancestor with upstream/develop: None                         
Ψ All dependencies are installed.                                       
Ψ Found arm-none-eabi-gcc version 10.1.0                                
Ψ Found avr-gcc version 8.5.0                                           
Ψ Found avrdude version 7.0                                             
Ψ Found dfu-programmer version 0.7.2                                    
Ψ Found dfu-util version 0.11                                           
Ψ Submodules are up to date.                                            
Ψ Submodule status:                                                     
Ψ - lib/chibios: 2023-04-15 13:48:04 +0000 --  (11edb1610)              
Ψ - lib/chibios-contrib: 2023-07-17 11:39:05 +0200 --  (da78eb37)       
Ψ - lib/googletest: 2021-06-11 06:37:43 -0700 --  (e2239ee6)            
Ψ - lib/lufa: 2022-08-26 12:09:55 +1000 --  (549b97320)                 
Ψ - lib/vusb: 2022-06-13 09:18:17 +1000 --  (819dbc1)                   
Ψ - lib/printf: 2022-06-29 23:59:58 +0300 --  (c2e3b4e)                 
Ψ - lib/pico-sdk: 2023-02-12 20:19:37 +0100 --  (a3398d8)               
Ψ - lib/lvgl: 2022-04-11 04:44:53 -0600 --  (e19410f8)                  
Ψ QMK is ready to go                                                    
```

> * https://get.vial.today/docs/firmware-size.html

Already used all options from this manual. I have enabled TAP_DANCE_ENABLE as in my setup I was geting 4 bytes free, probably will disable it again.

> * https://docs.qmk.fm/#/squeezing_avr

Thank you for this reference, I will take a close look on it.

**Comment by lesshonor** [comment]
> Looks like I use `avr-gcc` 8.5.0:

The QMK container [uses 8.3.0, it seems](https://github.com/qmk/qmk_base_container/blob/9b411f615dc851d5190e821585e6978268bff3ec/Dockerfile#L36).

I'd probably get rid of magic, space-cadet shifts and/or grave escape. Locking key support is also a pretty good option if it's still there; basically no one has locking switches (and even if they did, who would put one on a numpad?)

**Comment by OTLabs** [comment]
> The QMK container [uses 8.3.0, it seems](https://github.com/qmk/qmk_base_container/blob/9b411f615dc851d5190e821585e6978268bff3ec/Dockerfile#L36).

Oh, ok. Good to know! I can downgrade in due case or use the same container. Any plans to upgrade this container? ;-)

> I'd probably get rid of magic, space-cadet shifts and/or grave escape. Locking key support is also a pretty good option if it's still there; basically no one has locking switches (and even if they did, who would put one on a numpad?)

Thanks a lot for these tips!

**Comment by OTLabs** [comment]
Proposed changes led me to 2K extra free space!
``` * The firmware size is fine - 26572/28672 (92%, 2100 bytes free)```

File: keyboards/idobao/montex/v2/keymaps/vial/vial.json (hunk 1 above)
Comment (OTLabs) [reply keyboards/idobao/montex/v2/keymaps/vial/vial.json]: Implemented suggested changes

**Review by lesshonor — APPROVED** [review]
[Builds](https://github.com/lesshonor/vial-qmk/actions/runs/6235738716/job/16925612222#step:4:16) :)

> Any plans to upgrade this container? ;-)

That question would be better asked of the repository's maintainers, but I'm guessing a very important blocker is that [there is no 8.5.x for the Dockerfile to pull from ZakKemble's releases](https://github.com/ZakKemble/avr-gcc-build/releases).

**Comment by OTLabs** [comment]
Great news! Thanks a lot for your kind help!

**Pull request merged**